jobs/keepalived/templates/bin/utils/openstack/generate_token

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+#sourcing job properties to bring openstack credential in env variables
+# shellcheck disable=SC1091
+source /var/vcap/jobs/keepalived/bin/job_properties.sh
+jq=/var/vcap/packages/jq/bin/jq
+generate_body_for_token_req()
+{
+ cat << EOF
+{ "auth": {
+    "identity": {
+      "methods": ["password"],
+      "password": {
+        "user": {
+          "name": "${OS_USERNAME}",
+          "domain": { "name": "${OS_USER_DOMAIN_NAME}" },
+          "password": "${OS_PASSWORD}"
+        }
+      }
+    },
+    "scope": {
+      "project": {
+        "name": "${OS_PROJECT_NAME}",
+        "domain": { "name": "${OS_USER_DOMAIN_NAME}" }
+      }
+    }
+  }
+}
+EOF
+}
+generate_auth_token(){
+    local token_response
+    token_response=$(curl --retry 2 -sk -X POST \
```

Comment:
Not sure, will it also return a non zero status in case of a status code other than 201?
Wouldn't it be better to have http_status_code based check ?

Reply:
I think if we add the 'set -e', that would suffice. What do you think?

Reply:
As we are explicitly returning non-zero status code. Not breaking immediately as logging the error too.  So not adding set -e